src/rebuild/initiator.c

```diff
@@ -1046,9 +1141,12 @@ puller_cont_iter_cb(daos_handle_t ih, d_iov_t *key_iov,
 		}
 	} while (arg->re_iter);
 
-	rc = dc_cont_local_close(tls->rebuild_pool_hdl, coh);
-	if (rc)
-		return rc;
+close:
+	if (arg->snaps)
+		rebuild_snapshots_put(arg->snaps);
```

Comment:
So we know this must always be the last refcount, right? why do we need a refcount in first place?

Reply:
hmm, how?  puller_iter_arg is not per container, and there is no container tracking structure at all.

Reply:
Sorry, I didn't notice that rebuild_snapshots_put() is called in rebuild ULT on different xstreams, then please use some way to serialize 'ref++' & 'ref--'.

Reply:
probably I would just allocate space for each object, might less overhead in that way.